song-server/src/main/java/bio/overture/song/server/service/AnalysisTypeService.java

```diff
@@ -244,11 +255,82 @@ private void validateAnalysisTypeSchema(JsonNode analysisTypeSchema) {
     }
   }
 
+  private JsonNode mergeSchema(JsonNode oldSchema, JsonNode newSchema) {
```

Comment:
What is the purpose of this function? We weren't previously merging old schemas with new schemas, why is this needed now?

This function does not appear to have anything to do with AnalysisTypes other than that they are JsonNodes that need to be merged (if they do need to be merged...). Should this then be in a class for JSON utilties?

Lastly, is this code copied from somewhere else? This looks like a reusable method for merging JsonNodes, so if we did get this from somewhere else we should give it attribution to the original source. Please also update it to the coding format of the song codebase, for instance using lombok style variable declarations (`val`, `var`).